### PR TITLE
Improvements in qa_openstack

### PR DESCRIPTION
**Pull request by toabctl**

### CODE REVIEW

File: scripts/jenkins/qa_openstack.sh [hunk 1]
```diff
@@ -2,25 +2,26 @@
 # usage:
 # curl http://openqa.suse.de/sle/qatests/qa_openstack.sh | sh -x
 # needs 2.1GB space for /var/lib/{glance,nova}
-export MODE=kvm
-# Avoid nested virtualisation by setting the line below
-# Note that as of today (2014-02-14) OpenStack Nova has known
-# bugs. Since upstream tests with kvm, it doesn't really
-# make sense to test anything else.
-# MODE=lxc
-if ! grep -q -e vmx -e svm /proc/cpuinfo ; then
```
Comment (bmwiedemann) [inline scripts/jenkins/qa_openstack.sh]: should we try to error out early if someone requests KVM without these flags?

**Comment by bmwiedemann** [comment]
and what is "imrpo" in the PR title ? dont drink and code

File: scripts/jenkins/qa_openstack.sh [hunk 2]
```diff
@@ -19,8 +19,12 @@ ifconfig | grep inet
...
 # setup optional extra disk for cinder-volumes
 dev_cinder=/dev/vdb
-if ! test -e $dev_cinder && file -s /dev/sdb|grep -q "ext3 filesystem data" ; then
-    dev_cinder=/dev/sdb
+if ! test -e $dev_cinder ; then
+    # maybe we run under xen
+    dev_cinder=/dev/xvdb
+    if ! test -e $dev_cinder && file -s /dev/sdb|grep -q "ext3 filesystem data" ; then
+        dev_cinder=/dev/sdb
+    fi
```
Comment (aplanas) [inline scripts/jenkins/qa_openstack.sh]: I am ok with this, but I wonder why not to use a more simple approach here, like

```bash
if test -e /dev/vdb; then
  dev_cinder=/dev/vdb
elif ...
  ...
else
  echo "Extra disk for cinder-volumes not detected"
  exit -1 # if this an error?
fi
```

File: scripts/jenkins/qa_openstack.sh [hunk 3]
```diff
@@ -30,8 +34,13 @@ fi
...
 dev_manila=/dev/vdc
-if ! test -e $dev_manila && file -s /dev/sdc|grep -q "ext3 filesystem data" ; then
-    dev_manila=/dev/sdc
+if ! test -e $dev_manila ; then
+    # maybe we run under xen
+    dev_manila=/dev/xvdc
+
+    if ! test -e $dev_manila && file -s /dev/sdc|grep -q "ext3 filesystem data" ; then
+        dev_manila=/dev/sdc
+    fi
```
Comment (aplanas) [inline scripts/jenkins/qa_openstack.sh]: same here

File: scripts/jenkins/qa_openstack.sh (hunk 2 above)
Comment (toabctl) [inline scripts/jenkins/qa_openstack.sh]: Not sure if we want to error out if no extra disk is available.

File: scripts/jenkins/qa_openstack.sh (hunk 1 above)
Comment (toabctl) [inline scripts/jenkins/qa_openstack.sh]: Not sure. In the end MODE is only used for image selection.

File: scripts/jenkins/qa_openstack.sh [hunk 4]
```diff
@@ -138,38 +144,41 @@ if [ -n "$OSHEAD" ]; then
...
+        fi
     fi
-fi
 
-if [ "$VERSION" = "12" ] ; then
-    $zypper ar 'http://smt-internal.opensuse.org/repo/$RCE/SUSE/Products/SLE-SERVER/12/x86_64/product/' SLES12-Pool
-    $zypper ar -f 'http://smt-internal.opensuse.org/repo/$RCE/SUSE/Updates/SLE-SERVER/12/x86_64/update/' SLES12-Updates
-fi
+    if [ "$VERSION" = "12" ] ; then
+        $zypper ar 'http://smt-internal.opensuse.org/repo/$RCE/SUSE/Products/SLE-SERVER/12/x86_64/product/' SLES12-Pool
```
Comment (Itxaka) [inline scripts/jenkins/qa_openstack.sh]: Doesnt feel very DRY to have the same URL over and over here.

Comment (toabctl) [inline scripts/jenkins/qa_openstack.sh]: I have not changed that part. But I agree that this could be simplified.

**Pull request merged**